build/azure-pipelines/cli/cli-compile-and-publish.yml

```diff
@@ -0,0 +1,50 @@
+parameters:
+  - name: VSCODE_CLI_TARGET
+    type: string
+  - name: VSCODE_CLI_ARTIFACT
+    type: string
+  - name: VSCODE_CLI_ENV
+    type: object
+    default: {}
+
+steps:
+  - script: cargo build --release --target ${{ parameters.VSCODE_CLI_TARGET }} --bin=code-tunnel
+    displayName: Compile ${{ parameters.VSCODE_CLI_TARGET }}
+    workingDirectory: $(Build.SourcesDirectory)/cli
+    env:
+      VSCODE_CLI_VERSION: $(VSCODE_CLI_VERSION)
+      VSCODE_CLI_REMOTE_LICENSE_TEXT: $(VSCODE_CLI_REMOTE_LICENSE_TEXT)
+      VSCODE_CLI_REMOTE_LICENSE_PROMPT: $(VSCODE_CLI_REMOTE_LICENSE_PROMPT)
+      VSCODE_CLI_ASSET_NAME: ${{ parameters.VSCODE_CLI_ARTIFACT }}
+      VSCODE_CLI_AI_KEY: $(VSCODE_CLI_AI_KEY)
+      VSCODE_CLI_AI_ENDPOINT: $(VSCODE_CLI_AI_ENDPOINT)
+      ${{ each pair in parameters.VSCODE_CLI_ENV }}:
+        ${{ pair.key }}: ${{ pair.value }}
+
+  - ${{ if or(contains(parameters.VSCODE_CLI_TARGET, '-windows-'), contains(parameters.VSCODE_CLI_TARGET, '-darwin')) }}:
+    - task: ArchiveFiles@2
+      inputs:
+        ${{ if contains(parameters.VSCODE_CLI_TARGET, '-windows-')}}:
+          rootFolderOrFile: $(Build.SourcesDirectory)/cli/target/${{ parameters.VSCODE_CLI_TARGET }}/release/code-tunnel.exe
+        ${{ if contains(parameters.VSCODE_CLI_TARGET, '-darwin')}}:
```

Comment:
Should this be `elseif`? Or even `else`?

Reply:
I didn't know `else` was a thing in their conditionals!